server/service.go

```diff
@@ -129,6 +146,60 @@ func (m *BoostService) handleStatus(w http.ResponseWriter, req *http.Request) {
 	fmt.Fprintf(w, `{}`)
 }
 
+// sendValidatorPreferences is used to send the validators preferences to the registered relays
+func (m *BoostService) sendValidatorPreferences(log *logrus.Entry, payload []types.SignedValidatorRegistration) uint64 {
+	// We need a wait group to manage each routine used to perform the requests.
+	var wg sync.WaitGroup
+
+	// Use an atomic counter to count successful requests.
+	numSuccessRequestsToRelay := uint64(0)
+
+	// Send the validators preferences to each registered relay.
+	for _, relay := range m.relays {
+		wg.Add(1)
+
+		go func(relayAddr string) {
+			defer wg.Done()
+
+			url := relayAddr + pathRegisterValidator
+			log := log.WithField("url", url)
+
+			err := SendHTTPRequest(context.Background(), m.httpClient, http.MethodPost, url, payload, nil)
+			if err != nil {
+				log.WithError(err).Warn("error in registerValidator to relay")
+				return
+			}
+
+			atomic.AddUint64(&numSuccessRequestsToRelay, 1)
+		}(relay.Address)
+	}
+
+	wg.Wait()
+
+	return numSuccessRequestsToRelay
+}
+
+// registerValidatorAtInterval is used to update the registered validators preferences,
+// sending them to each relay at a given interval.
+//
+// Sending a value in the channel will stop the ticker.
+func (m *BoostService) registerValidatorAtInterval(interval time.Duration, done chan bool) {
+	log := m.log.WithField("method", "registerValidatorAtInterval")
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-done:
+			return
+		case <-ticker.C:
```

Comment:
One issue that arises is that we can send twice in a short interval (if a registration request comes near the timer tick)
In this approach it's not easy to solve it, and I think we could come at this from a different angle:
* make registerValidatorAtInterval the only place responsible for sending the registration
* listen on both the timer, and a new channel that accepts new registration requests
* on new registration request, send the preferences and reset the timer

This has some nice benefits:
* m.vp is now an object local to registerValidatorAtInterval
* validator preferences struct can become just a plain old map
* no need for mutex
* after a registration you will correctly wait until the next interval

The only downside is that you'll need to return the registration status back over a channel (or a callback), but it should be okay.

We can submit a PR to this PR if you prefer not implementing it

Reply:
Totally agree with this comment ! Thanks a lot for raising the issue and proposing an alternative 👍
I can make the updates don't worry.

Reply:
Added in de04d2f038dcde4efcd2bc8b7928911a762db457

cc @Ruteri